wsgi/prz-recipe-options.rst

```diff
@@ -23,30 +18,131 @@ The default is to listen on 0.0.0.0:8080.
 
 Use WSGI middleware debugging facilities.
+When set to ``on`` this option will disable exception views and thus propagate errors to the WSGI stack.
+This allows you to use specific debugging WSGI middleware like the `werkzeig debugger <https://werkzeug.palletsprojects.com/en/0.15.x/debug/>`_.
+We'll cover this in the Debugging chapter.
 
-http-fast-listen
-----------------
+access-log, z2-log
+------------------
+
+``access-log`` has been introduced as an alias for the ``z2-log`` option to specify the location of the access log file.
+
+access-log-level, z2-log-level
+------------------------------
+
+``access-log-level`` is a new alias for ``z2-log-level`` to specify the log level of the access log file.
+For WSGI the default is ``INFO`` (``WARN`` for ZServer).
+
+Advanced logging for WSGI
+-------------------------
+
+``access-log-handler``, ``access-log-args``, ``access-log-kwargs``, ``event-log-handler``, ``event-log-args``, ``event-log-kwargs`` are WSGI only options that you can use to configure arbitrary Python logging handlers.
+Numerous logging handlers are defined in the `Python standard library <https://docs.python.org/3/library/logging.handlers.html>`_.
+
+Exercise 1
+++++++++++
+
+Configure Plone access logging to a TCP Server using ``logging.handlers.SocketHandler`` from the Python standard library.
+Use a local address.
+
+
+..  admonition:: Solution
+    :class: toggle
+
+    Add the following two lines to your buildout configuration (we use ``options.cfg``) from the ``wsgitraining_buildout`` as a starting point:
+
+    .. code-block:: ini
+        :emphasize-lines: 9-10
+
+        ...
+        [instance]
+        recipe = plone.recipe.zope2instance
+        user = admin:admin
+        zeo-client = on
+        zeo-address = 8100
+        shared-blob = on
+        blob-storage = ${buildout:directory}/var/blobstorage
+        access-log-handler = logging.handlers.SocketHandler
+        access-log-args = ('localhost', 9000)
+        eggs =
+            Plone
+            Pillow
+            wsgitraining.site
+
+    ``plone.recipe.zope2instance`` uses default values for the ``access-log-args`` option, but not the ``access-log-kwargs`` option.
+    Same for the ``event-log-args/event-log-kwargs`` options.
+    This means you **have** to provide the ``...-log-args`` parameter, otherwise you will end up with the (in our case nonsensical) defaults in your ``wsgi.ini``.
+    After running buildout with ``buildout -c options.cfg`` you can start your instance with ``bin/instance fg``.
+    Use a tool like e.g. `netcat <http://netcat.sourceforge.net/>`_ (there is a package for your linux distribution) to open a listening socket: ``nc -l 9000``.
+    You will see the incoming log entries in pickled format when navigating to your Plone instance in the browser.
+
+sentry... options
+-----------------
+
+Sentry support for WSGI is  available through ``plone.recipe.zope2instance``.
+We'll cover these options later in the add-ons chapter.
```

Comment:
s/We'll/We will

Add cross-reference to add-ons chapter, for example:

```
doc:`filename-without-extension-for-add-ons-chapter`
```

Reply:
done also for reference to debugging chapter.